apps/web/src/app/accounts/accept-organization.component.ts

```diff
@@ -31,19 +31,29 @@ export class AcceptOrganizationComponent extends BaseAcceptComponent {
     platformUtilsService: PlatformUtilsService,
     i18nService: I18nService,
     route: ActivatedRoute,
-    private apiService: ApiService,
     stateService: StateService,
     private cryptoService: CryptoService,
     private policyApiService: PolicyApiServiceAbstraction,
     private policyService: PolicyService,
     private logService: LogService,
     private organizationApiService: OrganizationApiServiceAbstraction,
-    private organizationUserService: OrganizationUserService
+    private organizationUserService: OrganizationUserService,
+    private messagingService: MessagingService
   ) {
     super(router, platformUtilsService, i18nService, route, stateService);
   }
 
   async authedHandler(qParams: Params): Promise<void> {
+    const organizationInvitation = await this.stateService.getOrganizationInvitation();
+    if (organizationInvitation == null) {
+      // User has clicked an invitation link and is visiting this component for the first time
+      await this.unauthedHandler(qParams);
+      // Logging out the user to force the user to login again to evaluate if the user's current password meets the Organization password policy requirements
+      this.messagingService.send("logout", { redirect: false });
+      return;
+    }
```

Comment:
This is handled in a confusing way to me, we can update to remove the need to comment with some better naming
```suggestion
    const needsReAuth = (await this.stateService.getOrganizationInvitation()) != null;
    if (!needsReAuth) {
      // Accepting an org invite requires authentication from a logged out state
      this.messagingService.send("logout", { redirect: false });
      return;
    }
```

Also, it's not necessary to reference unauthHandler since if you log out, it'll handle as an unauth. To me, those two things really clear up what's happening here.

Reply:
Thanks for the suggestion, that does look simpler. I tried not referencing `unauthHandler` but it did not handle as an unauth, it only fires the first time and it's handled as authenticated. I've extracted the logic to a new method `prepareOrganizationInvitation` which then is called on both cases.